### PR TITLE
webrtc-audio-processing: 0.3.1 -> 1.0

### PR DESCRIPTION
**Pull request by viric**
###### Motivation for this change

Hoping for better AEC in pulseaudio recording with module-echo-cancel

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viric** [comment]
Oh. Unfortunately pulseaudio 14.2 doesn't pick it. I don't know why.

"checking for  webrtc-audio-processing >= 0.2 ... no"

File: pkgs/development/libraries/abseil-cpp/default.nix [hunk 1]
```diff
@@ -11,6 +11,8 @@ stdenv.mkDerivation rec {
     sha256 = "0dwxg54pv6ihphbia0iw65r64whd7v8nm4wwhcz219642cgpv54y";
   };
 
+  cmakeFlags = "-DBUILD_SHARED_LIBS=ON";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/abseil-cpp/default.nix:14]:
```suggestion
  cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ];
```

File: pkgs/development/libraries/webrtc-audio-processing/default.nix [hunk 2]
```diff
@@ -1,14 +1,17 @@
-{ lib, stdenv, fetchurl, darwin }:
+{ lib, stdenv, fetchurl, darwin, meson, abseil-cpp, ninja }:
 
 stdenv.mkDerivation rec {
-  name = "webrtc-audio-processing-0.3.1";
+  name = "webrtc-audio-processing-1.0";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/webrtc-audio-processing/default.nix:4]:
```suggestion
  pname = "webrtc-audio-processing";
  version = "1.0";
```

File: pkgs/development/libraries/webrtc-audio-processing/default.nix [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 
   src = fetchurl {
-    url = "https://freedesktop.org/software/pulseaudio/webrtc-audio-processing/${name}.tar.xz";
-    sha256 = "1gsx7k77blfy171b6g3m0k0s0072v6jcawhmx1kjs9w5zlwdkzd0";
+    url = "https://freedesktop.org/software/pulseaudio/webrtc-audio-processing/${name}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/webrtc-audio-processing/default.nix:7]:
```suggestion
    url = "https://freedesktop.org/software/pulseaudio/webrtc-audio-processing/${pname}-${version}.tar.gz";
```

File: pkgs/development/libraries/webrtc-audio-processing/default.nix [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
+    sha256 = "0vwkw5xw8l37f5vbzbkipjsf03r7b8nnrfbfbhab8bkvf79306j4";
   };
 
-  buildInputs = lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ ApplicationServices ]);
+  propagatedBuildInputs = [ abseil-cpp ];
+  buildInputs = [ meson ninja ] ++
+    lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks;
+      [ ApplicationServices ]);
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/webrtc-audio-processing/default.nix:14]:
```suggestion
  buildInputs = [ meson ninja ]
    ++ lib.optionals stdenv.isDarwin [ ApplicationServices ];
```

File: pkgs/development/libraries/webrtc-audio-processing/default.nix [hunk 5]
```diff
@@ -1,14 +1,17 @@
-{ lib, stdenv, fetchurl, darwin }:
+{ lib, stdenv, fetchurl, darwin, meson, abseil-cpp, ninja }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/webrtc-audio-processing/default.nix:1]:
```suggestion
{ lib, stdenv, fetchurl, darwin, meson, abseil-cpp, ninja, ApplicationServices }:
```

Please inherit ApplicationServices in top-level.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Please fix your commit message. It contained 0.1 instead of 1.0.

**Comment by viric** [comment]
I realized after submitting the PR that pulseaudio still cannot take the new code. And distributions package this as a different package; it has a different pkg-config name too... maybe we have to keep both then. I'll rework this.

**Pull request closed without merging**